### PR TITLE
Enabled or not default project-id.

### LINKED ISSUE #310: Latency during GcpCredentialRecorder loading
Hello, 

I have a latency of 3s during the startup of my quarkus application. This is related to the fact that the projectId is empty, so there is an attempt to retrieve the projectId by an http call from an internal google url with a timeout of 3s.

```
__  ____  __  _____   ___  __ ____  ______ 
--/ __ \/ / / / _ | / _ \/ //_/ / / / __/
-/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-07-28 11:18:33,334 CONFIG [com.goo.api.cli.htt.HttpTransport] (main) -------------- REQUEST  --------------
GET http://metadata.google.internal/computeMetadata/v1/project/project-id
Accept-Encoding: gzip
User-Agent: Google-HTTP-Java-Client/1.41.8 (gzip)
metadata-flavor: Google

2022-07-28 11:18:33,335 CONFIG [com.goo.api.cli.htt.HttpTransport] (main) curl -v --compressed -H 'Accept-Encoding: gzip' -H 'User-Agent: Google-HTTP-Java-Client/1.41.8 (gzip)' -H 'metadata-flavor: Google' -- 'http://metadata.google.internal/computeMetadata/v1/project/project-id'
2022-07-28 11:18:36,453 DEBUG [io.net.cha.DefaultChannelId] (Thread-1) -Dio.netty.processId: 124158 (auto-detected)

```
I can get the value of the projectId using a spring-cloud-server config, but the problem is that the quarkus-google-cloud-pubsub extension's recorder is executed before the "spring-cloud-server" extension's, Suddenly there is a NPE when starting the application because the attribute value is null (https://github.com/quarkiverse/quarkus-google-cloud-services/blob/1.2.0/common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpCredentialRecorder.java#L17).

```
quarkus:
  ...
  spring-cloud-config:
    enabled: true
    url: http://localhost:8888

  google:
    cloud:
      project-id: ${my-spring-cloud-config.gcp.projectid}
```

```
2022-07-28 14:14:14,052 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.lang.NullPointerException: Cannot invoke "java.util.Optional.isEmpty()" because "bootstrapConfig.projectId" is null
	at io.quarkiverse.googlecloudservices.common.GcpCredentialRecorder.configure(GcpCredentialRecorder.java:17)
	at io.quarkus.deployment.steps.CommonBuildSteps$configure370655783.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.CommonBuildSteps$configure370655783.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:108)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
```

If I put the projectid hard in the conf, it works fine.
Is it possible to get this extension to load after the config has been fully loaded? Or is there a workaround that I haven't thought of?

I used quarkus 2.11.1.RELEASE and quarkus-google-cloud-services 1.2.0.

Regards.

### PR DESCRIPTION
**Pull request by lucaspouzac**
Resolve #310

### CODE REVIEW

File: common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java [hunk 1]
```diff
@@ -13,9 +13,17 @@
  */
 @ConfigRoot(name = "google.cloud", phase = ConfigPhase.BOOTSTRAP)
 public class GcpBootstrapConfiguration {
+
+    /**
+     * Enabled set projectId by defaultV value, otherwise null.
+     *
+     */
+    @ConfigItem()
+    public boolean defaultProjectIdEnabled = true;
```
Comment (loicmathieu) [inline common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java:22]: I wonder if in the future we may want to use ServiceOptions to get "something" else, the Metadata server can retrieve other configuration options (like the region) and we may need it some day.

This to say that we may consider a different name for this option, like enableMetadataServer or loadDefaultFromMetadataServer or loadDefaultFromServiceOptions ...

Comment (lucaspouzac) [reply common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java:22]: ok, I agree with you.

File: common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java [hunk 2]
```diff
@@ -13,9 +13,16 @@
...
+    /**
+     * Enable or disable metadata server access to retrieve configuration options (projectId, region...).
+     */
+    @ConfigItem()
+    public boolean enableMetadataServer = true;
+
     /**
      * Google Cloud project ID.
-     * It defaults to `ServiceOptions.getDefaultProjectId()`,
+     * It defaults to `ServiceOptions.getDefaultProjectId()` if enableMetadataServer set tot true,
```
Comment (loicmathieu) [inline common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java:25]:
```suggestion
     * It defaults to `ServiceOptions.getDefaultProjectId()` if enableMetadataServer is set to true (which is the default),
```

**Review by loicmathieu — CHANGES_REQUESTED** [review]
Sorry, I forgot about this PR.
Can you make the small change suggested and squashed your commit ?

File: common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java (hunk 2 above)
Comment (lucaspouzac) [reply common/runtime/src/main/java/io/quarkiverse/googlecloudservices/common/GcpBootstrapConfiguration.java]: It's done! thanks

**Review by loicmathieu — APPROVED** [review]
LGTM, thanks

**Comment by loicmathieu** [comment]
@all-contributors please add @lucaspouzac for code

**Pull request merged**